src/test/scala/uk/gov/nationalarchives/pdi/test/QueryManagerSpec.scala

```diff
@@ -89,4 +97,65 @@ class QueryManagerSpec extends AnyWordSpec with Matchers with MockitoSugar {
     }
   }
 
+  "QueryManager.parseResultSet" must {
+    "return a non-empty RecordSet when ResultSet contains at least a row" in {
+
+      val mockResultSet = mock[ResultSet]
+      val mockQuerySolution = mock[QuerySolution]
+      val mockQuerySolution2 = mock[QuerySolution]
+      val mockRDFNodeVal1 = mock[RDFNode]
+      val mockRDFNodeVal2 = mock[RDFNode]
+      val mockRDFNodeVal3 = mock[RDFNode]
+      val mockRDFNodeVal4 = mock[RDFNode]
+
+      when(mockResultSet.hasNext) thenReturn true thenReturn true thenReturn false
+      when(mockResultSet.getResultVars) thenReturn newArrayList("col1", "col2")
+      when(mockResultSet.nextSolution()) thenReturn mockQuerySolution thenReturn mockQuerySolution2
+      when(mockResultSet.next()) thenReturn mockQuerySolution thenReturn mockQuerySolution2
+      when(mockQuerySolution.get("col1")) thenReturn mockRDFNodeVal1
+      when(mockQuerySolution.get("col2")) thenReturn mockRDFNodeVal2
+      when(mockRDFNodeVal1.asLiteral()) thenReturn ResourceFactory.createPlainLiteral("val1")
+      when(mockRDFNodeVal2.asLiteral()) thenReturn ResourceFactory.createPlainLiteral("val2")
+      when(mockQuerySolution2.get("col1")) thenReturn mockRDFNodeVal3
+      when(mockQuerySolution2.get("col2")) thenReturn mockRDFNodeVal4
+      when(mockRDFNodeVal3.asLiteral()) thenReturn ResourceFactory.createPlainLiteral("val3")
+      when(mockRDFNodeVal4.asLiteral()) thenReturn ResourceFactory.createPlainLiteral("val4")
+
+      val result = QueryManager.parseResultSet(mockResultSet)
```

Comment:
I don't think we should be testing `parseResultSet` as this is not the public API. Rather we should be testing `executeQueryAndGetRecordSet` in the same way that we test `executeQuery` elsewhere in this test suite.